beacon_node/beacon_chain/src/attestation_rewards.rs

```diff
@@ -49,12 +49,22 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             .get_state(&state_root, Some(state_slot))?
             .ok_or(BeaconChainError::MissingBeaconState(state_root))?;
 
+        let block_root = state
+            .get_block_root(state_slot)
+            .map_err(|err| BeaconChainError::BeaconStateError(err))?;
```

Comment:
This won't work, because the state doesn't contain the block root for the current slot (it would be a circular reference). This call will error every time.

Instead, you should use `state.get_latest_block_root(state_root)` to inject the state root and get the correct block root.